### PR TITLE
Containers: Separate engines_and_tools.yaml by architecture

### PR DESCRIPTION
**Pull request by pdostal**
To make the containers YAML more readable let's separate it by `ARCH`.

- [qac-openqa-yaml!296](https://gitlab.suse.de/qac/qac-openqa-yaml/-/merge_requests/296)
- Verification run: TBD

### CODE REVIEW

File: schedule/containers/engines_and_tools-s390x.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name:           engines_and_tools-s390x
+description:    >
+  Maintainer: qa-c@suse.de.
+  This schedule is focused on testing the related container packages and features,
+  but not the official container images from SUSE and openSUSE.
+conditional_schedule:
+  supported:
```
Comment (asmorodskyi) [inline schedule/containers/engines_and_tools-s390x.yaml:7]:
```suggestion
  buildah_podman:
```

**Review by asmorodskyi — APPROVED** [review]
I disagree but LGTM  :D

File: schedule/containers/engines_and_tools-s390x.yaml (hunk 1 above)
Comment (pdostal) [reply schedule/containers/engines_and_tools-s390x.yaml:7]: I would keep it as `supported` as also `registry` should be there as that test module is also not executed on every `VERSION`.

**Comment by b10n1k** [comment]
nice to have a "first" touch. a few points. 
- consider  https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12508. just a point for reviewers. 
- i would like to have a structure for the files as they tend to multiply fast. for instance taking your files for the example:
  ```
schedule/containers/engines_and_tools-aarch64.yaml -> schedule/containers/aarch64/engines_and_tools.yaml
schedule/containers/engines_and_tools-s390x.yaml -> schedule/containers/s390x/engines_and_tools.yaml
schedule/containers/engines_and_tools-x86_64.yaml -> schedule/containers/x86_64/engines_and_tools.yaml
```
Name can also be like `engines_and_tools@aarch64.yaml` if we like.

even better have a conventions like **schedule/containers/<testname>/(<common.yaml>|<arch>)/<aarch_specific>**. example:
```
schedule/containers/engines_and_tools-aarch64.yaml -> 
schedule/containers/engines_and_tools/aarch64/engines_and_tools.yaml
schedule/containers/engines_and_tools-s390x.yaml -> schedule/containers/engines_and_tools/s390x/engines_and_tools.yaml
schedule/containers/engines_and_tools-x86_64.yaml -> schedule/containers/engines_and_tools/engines_and_tools.yaml (assuming is used for ppc as well)
```
Directories can named based on our needs. 
opinions?

**Comment by pdostal** [comment]
> nice to have a "first" touch. a few points.
> 
> * consider  #12508. just a point for reviewers.
> * i would like to have a structure for the files as they tend to multiply fast. for instance taking your files for the example:
> 
> ```
> schedule/containers/engines_and_tools-aarch64.yaml -> schedule/containers/**aarch64**/engines_and_tools.yaml
> schedule/containers/engines_and_tools-s390x.yaml -> schedule/containers/**s390x**/engines_and_tools.yaml
> schedule/containers/engines_and_tools-x86_64.yaml -> schedule/containers/**x86_64**/engines_and_tools.yaml
> ```
> 
> Name can also be like `engines_and_tools@aarch64.yaml` if we like.
> 
> even better have a conventions like **schedule/containers / <common.yaml or arch> /<aarch_specific>/**. example:
> 
> ```
> schedule/containers/engines_and_tools-aarch64.yaml -> 
> schedule/containers/**engines_and_tools/aarch64**/engines_and_tools.yaml
> schedule/containers/engines_and_tools-s390x.yaml -> schedule/containers/**engines_and_tools/s390x**/engines_and_tools.yaml
> schedule/containers/engines_and_tools-x86_64.yaml -> schedule/containers/**engines_and_tools**/engines_and_tools.yaml (assuming is used for ppc as well)
> ```
> 
> Directories can named based on our needs.
> opinions?

I would keep the files out of directories and I would also avoid `@` in file names.

Of course if your docker / podman separation pull request will be merged I will rebase.

**Comment by jlausuch** [comment]
I also disagree, I still think having a new `main_containers.pm` would simplify many things to deal with these complex conditions for `VERSION` and `ARCH` at the same time.
However, my vote for this PR is neutral, so feel free to merge if you get approvals.

**Comment by pdostal** [comment]
Please leave open until something is decided.

**Review by grisu48 — APPROVED** [review]
LGTM :rocket: I'm in favour of moving forward with this PR.

**Comment by asmorodskyi** [comment]
> Please leave open until something is decided.

I vote for merging . like I said let's not create deadlocks like this . There are plenty of decisions in this repo which I don't like :) one more will not change a picture a lot :) You already have 2 approvals (including mine) and Jose is "not saying Yes but not saying NO either " so I am not sure what kind of "decision" you are waiting .
Please solve conflict and fix the CI and let's merge it

**Comment by pdostal** [comment]
Closing this pull request as it is being blocked for long time.

**Pull request closed without merging**